src/core_plugins/kbn_vislib_vis_types/public/controls/heatmap_options.html

```diff
@@ -0,0 +1,209 @@
+    <label class="kuiSideBarFormRow__label" for="axisScale">
+      Color Scale
+    </label>
+    <div class="kuiSideBarFormRow__control">
+      <select
+        id="axisScale"
+        class="kuiSelect kuiSideBarSelect"
+        ng-model="valueAxis.scale.type"
+        ng-options="mode for mode in vis.type.params.scales"
+      ></select>
+    </div>
+  </div>
+
+  <div class="kuiSideBarFormRow">
+    <label class="kuiSideBarFormRow__label" for="defaultYExtents">
+      Scale to Data Bounds
+    </label>
+    <div class="kuiSideBarFormRow__control">
+      <input class="kuiCheckBox" id="defaultYExtents" type="checkbox" ng-model="valueAxis.scale.defaultYExtents">
+    </div>
+  </div>
+
+  <div class="kuiSideBarFormRow" ng-if="!vis.params.setColorRange">
+    <label class="kuiSideBarFormRow__label" for="percentageMode">
+      Percentage Mode
+    </label>
+    <div class="kuiSideBarFormRow__control">
+      <input class="kuiCheckBox" id="percentageMode" type="checkbox" ng-model="vis.params.percentageMode">
+    </div>
+  </div>
+
+  <div class="kuiSideBarFormRow" ng-if="!vis.params.setColorRange">
+    <label class="kuiSideBarFormRow__label" for="colorsNumber">
+      Number of colors
+    </label>
+    <div class="kuiSideBarFormRow__control">
+      <input
+        id="colorsNumber"
+        class="kuiInput kuiSideBarInput"
+        ng-model="vis.params.colorsNumber"
+        type="number"
+        greater-than="1"
+        less-than="11"
+      >
+    </div>
+  </div>
+
+  <div>
+    <div class="kuiSideBarCollapsibleTitle">
+      <div
+        class="kuiSideBarCollapsibleTitle__label"
+        ng-click="vis.params.setColorRange = !vis.params.setColorRange"
+      >
+        <span
+          aria-hidden="true"
+          ng-class="{ 'fa-caret-down': vis.params.setColorRange, 'fa-caret-right': !vis.params.setColorRange }"
+          class="fa fa-caret-right kuiSideBarCollapsibleTitle__caret"
+        ></span>
+        <span class="kuiSideBarCollapsibleTitle__text">
+          Custom Ranges
+        </span>
+      </div>
+      <input aria-label="enable"
+             ng-model="vis.params.setColorRange"
+             type="checkbox"
+             class="kuiSideBarSectionTitle__action"
+      >
+    </div>
+
+    <div ng-if="vis.params.setColorRange" class="kuiSideBarCollapsibleSection">
+      <div class="kuiSideBarSection">
+        <table class="vis-editor-agg-editor-ranges form-group" ng-show="vis.params.colorsRange.length">
+          <tr>
+            <th></th>
+            <th>
+              <label>From</label>
+            </th>
+            <th colspan="2">
+              <label>To</label>
+            </th>
+          </tr>
+
+          <tr ng-repeat="range in vis.params.colorsRange track by $index">
+            <td>
+              <div class="kuiSideBarFormRow__label__colorbox" ng-style="{ 'background-color': getColor($index) }" />
+            </td>
+            <td>
+              <input
+                ng-model="range.from"
```

Comment:
Changing the colors via the legend and then specifying custom range's is behaving weirdly, maybe we should just remove the color box from when they specify custom ranges?

![heapmap-custom-colors](https://cloud.githubusercontent.com/assets/627123/21315566/d14dd4ca-c5ca-11e6-8c51-df1e8494da6c.gif)